File: dclib/core/src/main/java/dclib/util/CollectionUtils.java
package dclib.util;

import java.util.Map;

public final class CollectionUtils {

	private CollectionUtils() {
	}

	public static final <TKey, TValue> TValue get(final Map<TKey, TValue> map, final TKey key,
			final TValue defaultValue) {
		if (map.containsKey(key)) {
			return map.get(key);
		} else {
			map.put(key, defaultValue);
			return defaultValue;
		}
	}

}

File: dclib/core/src/main/java/dclib/util/CollectionUtils.kt
package dclib.util

import kotlin.reflect.KClass

object CollectionUtils {
    fun <TParent : Any, TChild : TParent> getByType(iterable: Iterable<TParent>, childClass: KClass<TChild>)
            : List<TChild> {
        return iterable.filter { it.javaClass == childClass.java }.map {
            @Suppress("UNCHECKED_CAST")
            it as TChild
        }
    }
}
